packages/broker/src/plugins/operator/OperatorValueBreachWatcher.ts

```diff
@@ -0,0 +1,59 @@
+import { EthereumAddress, Logger, scheduleAtInterval } from '@streamr/utils'
+import { MaintainOperatorValueHelper } from './MaintainOperatorValueHelper'
+import { OperatorServiceConfig } from './OperatorPlugin'
+
+const logger = new Logger(module)
+
+const CHECK_VALUE_INTERVAL = 1000 * 60 * 60 // 1 hour
+
+export class OperatorValueBreachWatcher {
+    private penaltyLimitFractionCached?: bigint
+    private readonly abortController: AbortController
+    
+    readonly helper: MaintainOperatorValueHelper
+
+    constructor(config: OperatorServiceConfig) {
+        this.helper = new MaintainOperatorValueHelper(config)
+        this.abortController = new AbortController()
+    }
+
+    async checkRandomUnwithdrawnEarnings(): Promise<void> {
+        const randomOperatorAddress = await this.helper.getRandomOperator()
+        logger.info('Checking unwithdrawn earnings', { randomOperatorAddress })
+        return this.checkUnwithdrawnEarningsOf(randomOperatorAddress)
+    }
+
+    async checkUnwithdrawnEarningsOf(targetOperatorAddress: EthereumAddress): Promise<void> {
+        logger.info('Check unwithdrawn earnings and check if they are above the safe threshold')
+        const { fraction, sponsorshipAddresses } = await this.helper.getUnwithdrawnEarningsOf(targetOperatorAddress)
+        const limit = await this.getPenaltyLimitFraction()
+        logger.info(` -> is ${fraction} > ${limit}?`)
```

Comment:
Looks like a development-time debug statement. Remove, or move to trace level:
```suggestion
        logger.trace(` -> is ${fraction} > ${limit}?`)
```

Reply:
Changed